boring/src/x509/mod.rs

```diff
@@ -157,6 +159,28 @@ impl X509StoreContextRef {
         unsafe { cvt_n(ffi::X509_verify_cert(self.as_ptr())).map(|n| n != 0) }
     }
 
+    /// Verifies the stored certificate with additional untrusted CRLs
+    ///
+    /// Returns `true` if verification succeeds. The `error` method will return the specific
+    /// validation error if the certificate was not valid.
+    ///
+    /// This will only work inside of a call to `init`.
+    ///
+    /// This corresponds to [`X509_STORE_CTX_set0_crls`] followed by [`X509_verify_cert`].
+    ///
+    /// [`X509_STORE_CTX_set0_crls`]:  https://www.openssl.org/docs/man1.0.2/crypto/X509_STORE_CTX_set0_crls.html
+    /// [`X509_verify_cert`]:  https://www.openssl.org/docs/man1.0.2/crypto/X509_verify_cert.html
+    pub fn verify_cert_with_crls(
+        &mut self,
+        untrusted_crls: Stack<X509CRL>,
+    ) -> Result<bool, ErrorStack> {
+        unsafe {
+            ffi::X509_STORE_CTX_set0_crls(self.as_ptr(), untrusted_crls.as_ptr());
+            mem::forget(untrusted_crls);
+            cvt_n(ffi::X509_verify_cert(self.as_ptr())).map(|n| n != 0)
+        }
+    }
```

Comment:
So, funnily enough, in my original commit I had this in a separate setter. But the first thing `init` does is [zero out everything in context](https://github.com/google/boringssl/blob/56eeb201aeafc2354d6f172c1695fffca4a3dd04/crypto/x509/x509_vfy.c#L2253), including the pointer to the crl stack, which was bad. It makes it quite hard to use as a separate function, since you need to make sure to call `set_crls` within the verify function passed to `init`. 

Anyways, I assumed that the zeroing meant that either verify or `X509_STORE_CTX_cleanup` must free the crl stack. But now I can't find anywhere that actually frees the crl stack in boringssl 😬 (and it sort of looks like after the call the pointer is still live in the debugger). So despite this being a `set0` it looks like ownership may stay with the caller (and this as written is a memory leak).

I'll fix that, but open to any suggestions on how to clean up the API... I can at least call `X509_STORE_CTX_set0_crls(nullptr)` afterwords if I can't think of anything better. An alternative I considered was to add a parameter to `init`, however that would break existing callers (so would require a new method with a new name).

Reply:
> It could still be a separate method that you have to call while within init

My weak argument against this: It's hard to use correctly because if 
My strong argument: Because `set0` isn't taking ownership, you have to call `verify` while your `Stack` is in scope to avoid an invalid memory access:
```rust
ctx.init(foo,bar,baz, |c| {
  if (have_crls) {
     stack = copy_my_crls_into_a_stack(crlsVec);
     c.set_crls(stack);
  } // <-- stack is dropped, ctx's crls pointer is bunk
  c.verify(); // 💣 
});
```

pushed a change that `null`s out the pointer at least!